docs/DeveloperGuide.md

```diff
@@ -1,34 +1,670 @@
+    * If the user has included a title in the command, the `findTitle` function would go through the list to 
+find the title. If the input title does not match any of the titles in the list, the command would terminate with 
+a message: “title is not found”. If a match is found, the system prints out the note and asks for confirmation 
+for deletion. In the case of multiple matches, the system would print all matching cases and ask the user to select 
+amongst them. Following the confirmation, a “N” would terminate the command while a “Y” would proceed to remove 
+the note from the list.
+    * If the user did not pre-input the title, the system would print all notes currently in the list and ask 
+for the user to select which note to delete among them. Following the confirmation for deletion, a “N” would terminate 
+the command while a “Y” would proceed to remove the note from the list.
+
+1. The current note list would then be saved by the `Storage` class.
+
+Figure 4.5-2 illustrates the above steps via a Sequence Diagram.
+
+![Figure 4.5-2](images/DeveloperGuide/Figure 4.5-2.png)
+_Figure 4.5-2: Sequence diagram for deleteNotesCommand_
+
+##### Design Considerations
+
+* Any number inputs by the user must be checked through to ensure that it is not out of the available indexes 
+in the array. 
+* The function must deal with incorrect title inputs by the user. The user cannot input a title that is not found 
+in any of the notes nor can he input an invalid title such as an empty string.
+
+* In the event of an empty list, the user cannot delete any more notes from the list. Hence this would result in 
+an exception caught.
+
+### 4.6 Undoing Changes Made to Tasks and Notes (Johannine)
+
+The undo feature allows the user to undo any changes made to Task or Note objects, particularly edits and deletions.
+
+##### Implementation
+
+To implement the `undo` feature, the concept of a stack was used to hold all the history of previous versions of `Tasks` 
+(or `Notes`) before they are changed.
+
+At every instance where a particular `Task` (or `Note`) is edited or deleted, using commands such as `editDeadline`, 
+`deleteTask` or `editNote`, a copy of the `Task` (or `Note`) is made as the changes are being made. Every `Task` or 
+`Note` object has an `editNumber` attributed to it, which is assigned a positive value if it has been edited, and a 
+negative value if it has been deleted.
+
+The copy made is temporarily stored as a new `Task` (or `Note`) object until the edit or deletion is successful. The 
+copy of the old unchanged `Task` (or `Note`) is then pushed into an array called `taskHistory` (or `NoteHistory`), 
+which holds all the previous copies of the object.
+
+Figure 4.6-1 illustrates the sequence diagram of the concept above, applied on changes made to a `Task`. The concept 
+works in a similar manner for that of `Note` objects.
+
+![Figure 4.6-1](images/DeveloperGuide/Figure 4.6-1.png)
```

Comment:
The self-invocation marked as an assignment (the equal to sign) is a bit unclear (highlighted in yellow)
Also, should there be a dotted line after the Task constructor.
![image](https://user-images.githubusercontent.com/60306071/97668982-927fc700-1a9c-11eb-991c-d9f4cd2f9d3e.png)